### ISSUE TITLE
Crash Upon Entering Fight with 0 HP

### ISSUE THREAD

**Issue by Icyvexen** [TerryCavanagh/diceydungeons.com#1317]
# Lead-up

- Warrior - Episode 2
- 4th Floor fight against Jester
- Jester had thorns, final attack on him left both Jester and Warrior with 0 health
- Jester finished his dialogue, was still alive with 0 health
- Upon attempting to encounter the next fight, game crashes

# Crash Log:

```txt
--------------------------------------
filename:	diceydungeons
package:	com.terrycavanaghgames.diceydungeons
version:	v1.2 / 0.11.0 (mod API)
sess. ID:	dicey_dungeons_2019-08-25_16'33'47
started:	2019-08-25 16:33:47
--------------------------------------
crashed:	2019-08-25 18:56:51
duration:	02:23:03
error:		Null Object Reference
stack:
elements.Floor.activatenode (elements/Floor.hx line 1147)
elements.Floor.playerstep (elements/Floor.hx line 910)
Reflect.callMethod (C:\coding\haxe\haxe\std/cpp/_std/Reflect.hx line 55)
motion.actuators.GenericActuator.complete (motion/actuators/GenericActuator.hx line 155)
motion.actuators.SimpleActuator.update (motion/actuators/SimpleActuator.hx line 501)
motion.actuators.SimpleActuator.stage_onEnterFrame (motion/actuators/SimpleActuator.hx line 579)
openfl.events.EventDispatcher.__dispatchEvent (openfl/events/EventDispatcher.hx line 443)
openfl.display.DisplayObject.__dispatch (openfl/display/DisplayObject.hx line 1236)
```

# Picture if it helps

![image](https://user-images.githubusercontent.com/13083288/63657793-e4c3aa80-c76a-11e9-9485-a06d47a2b5d7.png)

**Comment by omor1** [comment]
I parse this as two interacting bugs:

- The game expects health > 0 on entering a fight and crashes when that's not the case
- You should lose if your health ever goes to zero (except _maybe_ against the boss)

**Comment by TerryCavanagh** [comment]
Fixed in v1.9!
